geode-core/src/main/java/org/apache/geode/cache/control/RestoreRedundancyResults.java

```diff
@@ -0,0 +1,148 @@
+   * Returns the {@link RegionRedundancyStatus} for a specific region or null if that region
+   * is not present in this {@link RestoreRedundancyResults}.
+   *
+   * @param regionName The region to which the {@link RegionRedundancyStatus} to be returned
+   *        belongs.
+   * @return A {@link RegionRedundancyStatus} for the specified region or null if that region is not
+   *         present in this {@link RestoreRedundancyResults}.
+   */
+  RegionRedundancyStatus getRegionResult(String regionName);
+
+  /**
+   * Returns all the {@link RegionRedundancyStatus RegionRedundancyStatuses} for regions with
+   * configured redundancy but zero actual redundant copies.
+   *
+   * @return A {@link Map} of {@link String} region name to {@link RegionRedundancyStatus} for every
+   *         region contained in this {@link RestoreRedundancyResults} with configured redundancy
+   *         but zero actual redundant copies.
+   */
+  Map<String, RegionRedundancyStatus> getZeroRedundancyRegionResults();
+
+  /**
+   * Returns all the {@link RegionRedundancyStatus RegionRedundancyStatuses} for regions with with
+   * at least one redundant copy, but fewer than the configured number of redundant copies.
+   *
+   * @return A {@link Map} of {@link String} region name to {@link RegionRedundancyStatus} for every
+   *         region contained in this {@link RestoreRedundancyResults} with at least one redundant
+   *         copy, but fewer than the configured number of redundant copies.
+   */
+  Map<String, RegionRedundancyStatus> getUnderRedundancyRegionResults();
+
+  /**
+   * Returns all the {@link RegionRedundancyStatus RegionRedundancyStatuses} for regions with
+   * redundancy satisfied.
+   *
+   * @return A {@link Map} of {@link String} region name to {@link RegionRedundancyStatus} for every
+   *         region contained in this {@link RestoreRedundancyResults} with redundancy satisfied.
+   */
+  Map<String, RegionRedundancyStatus> getSatisfiedRedundancyRegionResults();
+
+  /**
+   * Returns all the {@link RegionRedundancyStatus RegionRedundancyStatuses} contained in this
+   * {@link RestoreRedundancyResults}. This method may return the actual backing map depending on
+   * implementation.
+   *
+   * @return A {@link Map} of {@link String} region name to {@link RegionRedundancyStatus} for every
+   *         region contained in this {@link RestoreRedundancyResults}.
+   */
+  Map<String, RegionRedundancyStatus> getRegionResults();
+
+  /**
+   * Returns the total number of primaries that were transferred as part of the restore redundancy
+   * operations.
+   *
+   * @return the total number of primaries that were transferred
+   */
+  int getTotalPrimaryTransfersCompleted();
+
+  /**
+   * Returns the total time, in milliseconds, spent transferring primaries as part of the restore
+   * redundancy operations.
+   *
+   * @return the total time, in milliseconds, spent transferring primaries
+   */
+  long getTotalPrimaryTransferTime();
```

Comment:
We had discussion long ago about making sure time based APIs used `TimeUnit`. This method should be `long getTotalPrimaryTransferTime(TimeUnit unit)`. Alternatively we could consider `java.time.Duration`. Although that does incur an object allocation, given the infrequency that it is called the overhead is probably worth the correctness.

Reply:
Done